collect_app/src/main/java/org/odk/collect/android/widgets/AbstractImageWidget.java

```diff
@@ -17,10 +17,15 @@
 package org.odk.collect.android.widgets;
 
 import android.content.Context;
+import android.support.annotation.Nullable;
+import android.widget.ImageView;
 
 import org.javarosa.form.api.FormEntryPrompt;
 
 public abstract class AbstractImageWidget extends QuestionWidget {
+    @Nullable
+    protected ImageView imageView;
```

Comment:
Agreed! Getting to the most useful level of granularity is certainly an art. In this case I think it would be nice to have all commits up to 1b61ef4 squashed into one with a message like "Pull duplicated code to abstract base class". @grzesiek2010 you can do that with `rebase -i` and a series of squashes. There are some later commits that should also be included in that one. You can move them up and squash them with `rebase -i` as well.

If this were a smaller change I would squash the whole thing on merge but I think it will be helpful for the future to separate out the refactor from the fix.